655 migration override userprofile

selimnairb opened the pull request:
This PR updates the ingest code so that after resource creation, we overwrite the owner's creator information for the resource with what was in the bag (back when the resource was last edited), rather than what is in the user profile (at the time of ingest).

selimnairb commented:
@pkdash @hyi Can you review?  @mjstealey Can we try this in another dry run tomorrow (Friday the 9th)?

selimnairb commented on hs_core/serialization.py:
```diff
@@ -681,22 +673,59 @@ def write_metadata_to_resource(self, resource):
+
+                    # Find the owner in the creators metadata
+                    owner_metadata = None
+                    for rc in resource.metadata.creators.all():
```
@hyi Ideally I would query Django for the creator whose order == 1, but I couldn't quickly figure out how to do this with a GenericRelation.  Do you have any idea?

hyi replied:
@selimnairb Could you do owner_metadata = resource.metadata.creators.filter(order=1).first() ?

selimnairb commented:
@hyi Thanks for the suggestion to use filter instead of a linear search.  @pkdash Can you review when you have a chance?  We'd like to do another dry run this afternoon.  Thanks.

selimnairb commented:
@pkdash The ingestion code does not currently try to read researcherID or researchGateID from the XML because I didn't have an example resource with these fields to test with.  I don't believe it is currently possible to create these fields in the web interface, but if it is possible, please let me know how and I will add this.

selimnairb commented:
@pkdash I fixed the researchGageID typo.  I also fixed a limitation in the SAX parser (used to ensure some elements are parsed in the order encountered in the resourcemetadata.xml).  Please let me know if this can be merged.  Thanks.

pkdash commented:
+1
